packages/console/src/pages/EnterpriseSsoDetails/Connection/SamlMetadataForm/ParsedConfigPreview/index.tsx

```diff
@@ -15,51 +15,62 @@ type Props = {
   identityProviderConfig: ParsedSsoIdentityProviderConfig<SsoProviderName.SAML>['identityProvider'];
 };
 
+type CertificatePreviewProps = {
+  identityProviderConfig: { x509Certificate: string; expiresAt: number; isValid: boolean };
+  className?: string;
+};
+
+export function CertificatePreview({
+  identityProviderConfig: { x509Certificate, expiresAt, isValid },
+  className,
+}: CertificatePreviewProps) {
+  const { language } = i18next;
+  return (
+    <div className={classNames(styles.certificatePreview, className)}>
+      <div className={classNames(styles.indicator, !isValid && styles.errorStatus)} />
+      <DynamicT
+        forKey="enterprise_sso_details.saml_preview.certificate_content"
+        interpolation={{
+          date: new Date(expiresAt).toLocaleDateString(
+            // TODO: check if Logto's language tags are compatible.
+            conditional(isLanguageTag(language) && language) ?? 'en',
+            {
+              weekday: 'long',
+              year: 'numeric',
+              month: 'long',
+              day: 'numeric',
+            }
+          ),
+        }}
```

Comment:
nit: Can we use dayjs to format the date? Seems it would be easier that way?

Reply:
Check date-fns usage later, this need a mapping between logto and date-fns locales.